### ISSUE TITLE
UNMET PEER DEPENDENCY on installing @angular/material

### ISSUE THREAD

**Issue by drasko** [angular/material2#1512]
#### Bug, feature request, or proposal:
Can not install Material 2

#### What is the current behavior?
```
drasko@Lenin:~/mainflux/dashflux$ npm install --save @angular/material
dashflux@0.0.0 /home/drasko/mainflux/dashflux
├── UNMET PEER DEPENDENCY @angular/common@2.0.2
├── UNMET PEER DEPENDENCY @angular/compiler@2.0.2
├── UNMET PEER DEPENDENCY @angular/core@2.0.2
├─┬ @angular/material@2.0.0-alpha.9-3 
│ └── @types/hammerjs@2.0.33 
└── UNMET PEER DEPENDENCY @angular/platform-browser@2.0.2

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.14: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN @angular/compiler-cli@0.6.4 requires a peer of @angular/platform-server@2.0.2 but none was installed.
npm WARN @angular/platform-server@2.1.0 requires a peer of @angular/core@2.1.0 but none was installed.
npm WARN @angular/platform-server@2.1.0 requires a peer of @angular/common@2.1.0 but none was installed.
npm WARN @angular/platform-server@2.1.0 requires a peer of @angular/compiler@2.1.0 but none was installed.
npm WARN @angular/platform-server@2.1.0 requires a peer of @angular/platform-browser@2.1.0 but none was installed.
drasko@Lenin:~/mainflux/dashflux$
```

#### Which versions of Angular, Material, OS, browsers are affected?
```
drasko@Lenin:~/mainflux/dashflux$ uname -a
Linux Lenin 4.6.0-1-amd64 #1 SMP Debian 4.6.4-1 (2016-07-18) x86_64 GNU/Linux
drasko@Lenin:~/mainflux/dashflux$ node --version
v6.8.1
drasko@Lenin:~/mainflux/dashflux$ ng --version
Looks like you have a different program called watchman, falling back to NodeWatcher.
Visit http://ember-cli.com/user-guide/#watchman for more info.
angular-cli: 1.0.0-beta.17
node: 6.8.1
os: linux x64
```

**Comment by JanStureNielsen** [comment]
It looks like you're mixing versions, @drasko; here are the relevant portions of my `package.json` which is running Angular 2.1.0 with Angular Material 2.0.0-alpha.9-3 successfully on Angular CLI 1.0.0-beta.17:

```
  "dependencies": {
    "@angular/common": "2.1.0",
    "@angular/compiler": "2.1.0",
    "@angular/core": "2.1.0",
    "@angular/forms": "2.1.0",
    "@angular/http": "2.1.0",
    "@angular/platform-browser": "2.1.0",
    "@angular/platform-browser-dynamic": "2.1.0",
    "@angular/router": "3.1.0",
    "@angular/material": "2.0.0-alpha.9-3",
    "core-js": "^2.4.1",
    "rxjs": "5.0.0-beta.12",
    "ts-helpers": "^1.1.1",
    "zone.js": "^0.6.23"
  },
  "devDependencies": {
    "@types/jasmine": "^2.2.30",
    "@types/node": "^6.0.42",
    "hammerjs": "2.0.8",
    "angular-cli": "1.0.0-beta.17",
    "codelyzer": "~0.0.26",
    "jasmine-core": "2.4.1",
    "jasmine-spec-reporter": "2.5.0",
    "karma": "1.2.0",
    "karma-chrome-launcher": "^2.0.0",
    "karma-cli": "^1.0.1",
    "karma-jasmine": "^1.0.2",
    "karma-remap-istanbul": "^0.2.1",
    "protractor": "4.0.9",
    "ts-node": "1.2.1",
    "tslint": "3.13.0",
    "typescript": "2.0.3"
  }

```

**Comment by drasko** [comment]
`package.json` was generated with `angular-cli`.

```json
drasko@Lenin:~/mainflux/dashflux$ cat package.json 
{
  "name": "dashflux",
  "version": "0.0.0",
  "license": "MIT",
  "angular-cli": {},
  "scripts": {
    "start": "ng serve",
    "lint": "tslint \"src/**/*.ts\"",
    "test": "ng test",
    "pree2e": "webdriver-manager update",
    "e2e": "protractor"
  },
  "private": true,
  "dependencies": {
    "@angular/common": "~2.0.0",
    "@angular/compiler": "~2.0.0",
    "@angular/core": "~2.0.0",
    "@angular/forms": "~2.0.0",
    "@angular/http": "~2.0.0",
    "@angular/material": "^2.0.0-alpha.9-3",
    "@angular/platform-browser": "~2.0.0",
    "@angular/platform-browser-dynamic": "~2.0.0",
    "@angular/router": "~3.0.0",
    "core-js": "^2.4.1",
    "rxjs": "5.0.0-beta.12",
    "ts-helpers": "^1.1.1",
    "zone.js": "^0.6.23"
  },
  "devDependencies": {
    "@types/jasmine": "^2.2.30",
    "@types/node": "^6.0.42",
    "angular-cli": "1.0.0-beta.17",
    "codelyzer": "~0.0.26",
    "jasmine-core": "2.4.1",
    "jasmine-spec-reporter": "2.5.0",
    "karma": "1.2.0",
    "karma-chrome-launcher": "^2.0.0",
    "karma-cli": "^1.0.1",
    "karma-jasmine": "^1.0.2",
    "karma-remap-istanbul": "^0.2.1",
    "protractor": "4.0.9",
    "ts-node": "1.2.1",
    "tslint": "3.13.0",
    "typescript": "2.0.2"
  }
}
```

And then I did `npm install --save @angular/material`

Is this a bug in `angular-cli`?

**Comment by keyduq** [comment]
I think this is an angular-cli bug [#2692](https://github.com/angular/angular-cli/issues/2692)

**Comment by drasko** [comment]
@JanStureNielsen trying `package.json` you proposed, I am still getting the errors:

```
drasko@Lenin:~/mainflux/dashflux$ npm install --save @angular/material
dashflux@0.0.0 /home/drasko/mainflux/dashflux
├── UNMET PEER DEPENDENCY @angular/compiler@2.1.0
├── UNMET PEER DEPENDENCY @angular/core@2.1.0
└── @angular/material@2.0.0-alpha.9-3 

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.14: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN @angular/compiler-cli@0.6.4 requires a peer of @angular/compiler@2.0.2 but none was installed.
npm WARN @angular/compiler-cli@0.6.4 requires a peer of @angular/platform-server@2.0.2 but none was installed.
npm WARN @angular/compiler-cli@0.6.4 requires a peer of @angular/core@2.0.2 but none was installed.
npm WARN @ngtools/webpack@1.1.0 requires a peer of typescript@2.0.2 but none was installed.
```

**Comment by jelbourn** [comment]
Try using the same package.json from this sample app: https://github.com/angular/material2

In that, I don't get any unmet peer deps for material itself, just for compiler-cli (which is unrelated to material). In general, though, you can ignore the unmet peer dependencies.

**Comment by pliski** [comment]
After updating to angular-cli@1.0.0-beta.18 and @angular/material@2.0.0-alpha.9-3 I don't get anymore "unmet peer dependency" messages.

**Comment by elyas74** [comment]
same issue :
```
➜  testing git:(master) npm install --save @angular/material
testing@0.0.0 /home/elyas/programming/js/testing
├── UNMET PEER DEPENDENCY @angular/common@4.0.1
├── UNMET PEER DEPENDENCY @angular/core@4.0.1
├── UNMET PEER DEPENDENCY @angular/http@4.0.1
└── @angular/material@2.0.0-beta.2 

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN @angular/material@2.0.0-beta.2 requires a peer of @angular/core@^2.3.0 but none was installed.
npm WARN @angular/material@2.0.0-beta.2 requires a peer of @angular/common@^2.3.0 but none was installed.
npm WARN @angular/material@2.0.0-beta.2 requires a peer of @angular/http@^2.3.0 but none was installed.
```
@angular/cli: 1.0.0
node: 6.10.0
os: linux x64

**Comment by JordyBaylac** [comment]
same issue as @elyas74

**Comment by Chan4077** [comment]
@elyas74 @JordyBaylac Try installing from master:
```bash
npm install --save https://github.com/angular/material2-builds.git
```

**Comment by cmmora** [comment]
hi

maybe i have the same issue

npm install --save https://github.com/angular/material2-builds.git
angular-demo@0.0.0 /storage/cmora/Solucionet/proyectos/angular/angularDemo
├── UNMET PEER DEPENDENCY @angular/cdk@2.0.0-beta.8
├── UNMET PEER DEPENDENCY @angular/common@2.4.10
├── UNMET PEER DEPENDENCY @angular/core@2.4.10
└── @angular/material@2.0.0-beta.8-abec992  (git+https://github.com/angular/material2-builds.git#03dd33fbfdc7948c7da7d4676b0ed4ec7e081c47)

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN @angular/cdk@2.0.0-beta.8 requires a peer of @angular/core@^4.0.0 but none was installed.
npm WARN @angular/cdk@2.0.0-beta.8 requires a peer of @angular/common@^4.0.0 but none was installed.
npm WARN @angular/material@2.0.0-beta.8-abec992 requires a peer of @angular/cdk@2.0.0-beta.8-abec992 but none was installed.
npm WARN @angular/material@2.0.0-beta.8-abec992 requires a peer of @angular/core@^4.0.0 but none was installed.
npm WARN @angular/material@2.0.0-beta.8-abec992 requires a peer of @angular/common@^4.0.0 but none was installed.


angular-cli: 1.0.0-beta.28.3
node: 6.11.1
os: linux x641.1

**Comment by Chan4077** [comment]
@cmmora What do you expect when you're using an **older** version! From a certain version, it requires angular 4 and above, which explains the error as shown above.

**Comment by Saifeddine-Rjab** [comment]
solved for me with installing the 6.4.7 version
npm install @angular/material@6.4.7 --save
